loki/transformations/inline/functions.py

```diff
@@ -235,6 +234,13 @@ def inline_statement_functions(routine):
                 spec_map[decl] = None
     routine.spec = Transformer(spec_map).visit(routine.spec)
 
+def _get_callee_result_var(callee):
+    if callee.result_name.lower() in callee.variable_map:
+        callee_result_var = callee.variable_map[callee.result_name.lower()]
+    else:
+        callee_result_var_type = callee.symbol_attrs[callee.result_name.lower()]
+        callee_result_var = sym.Variable(name=callee.result_name, type=callee_result_var_type, scope=callee)
```

Comment:
NB: these are all case insensitive dicts, i.e. you can avoid the lower() conversion:
```suggestion
    if callee.result_name in callee.variable_map:
        callee_result_var = callee.variable_map[callee.result_name]
    else:
        callee_result_var_type = callee.symbol_attrs[callee.result_name]
        callee_result_var = sym.Variable(name=callee.result_name, type=callee_result_var_type, scope=callee)
```